dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/JsonObjectDeserializer.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence
+
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.log.internal.utils.errorWithTelemetry
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import com.google.gson.JsonParser
+import java.util.Locale
+
+internal class JsonObjectDeserializer : Deserializer<String, JsonObject> {
+    override fun deserialize(model: String): JsonObject? {
+        return try {
+            JsonParser.parseString(model).asJsonObject
+        } catch (jpe: JsonParseException) {
+            sdkLogger.errorWithTelemetry(
```

Comment:
I am afraid that if we have a problem with one of our formats in one big customer we will overload the telemetry here. I know you have your back pressure mechanism in the telemetry handler just wanted to make sure you considered this.

Reply:
It is not different from what we have in any `Deserializer` right now, all of them also submit such telemetry. So far no issues spotted :)